pkg/controller/master/storagenetwork/storage_network.go

```diff
@@ -298,34 +300,55 @@ func (h *Handler) createNad(setting *harvesterv1.Setting) (string, error) {
 	nad.Annotations = map[string]string{
 		StorageNetworkAnnotation: "true",
 	}
+	nad.Labels = map[string]string{
+		HashStorageNetworkLabel: h.sha1(setting.Value),
+	}
 	nad.Spec.Config = string(nadConfig)
 
 	// create nad
 	var nadResult *nadv1.NetworkAttachmentDefinition
 	if nadResult, err = h.networkAttachmentDefinitions.Create(&nad); err != nil {
-		return "", fmt.Errorf("create net-attach-def failed %v", err)
+		return nil, fmt.Errorf("create net-attach-def failed %v", err)
+	}
+
+	return nadResult, nil
+}
+
+func (h *Handler) findOrCreateNad(setting *harvesterv1.Setting) (*nadv1.NetworkAttachmentDefinition, error) {
+	sets := labels.Set{
+		HashStorageNetworkLabel: h.sha1(setting.Value),
+	}
+
+	nads, err := h.networkAttachmentDefinitionsCache.List(StorageNetworkNetAttachDefNamespace, sets.AsSelector())
```

Comment:
What if the cache is stale, and we still get no NAD in return? Maybe we can use the client to list. Though expensive, we won't do this often. It should be fine.

Reply:
Updated per suggestion, thanks.